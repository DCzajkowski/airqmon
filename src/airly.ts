export const AIRLY_API_URL = 'https://airapi.airly.eu';

export interface IAirlyCurrentMeasurement {
  airQualityIndex: number;
  pm1: number;
  pm25: number;
  pm10: number;
  pressure: number;
  humidity: number;
  temperature: number;
  pollutionLevel: number;
}
export interface IArilyNearestSensorMeasurement {
  airQualityIndex: number;
  pm25: number;
  pm10: number;
  distance: number;
  id: number;
  name: string;
  vendor: string;
  location: {
    latitude: number;
    longitude: number;
  };
  pollutionLevel: number;
  address: {
    streetNumber: number;
    route: string;
    locality: string;
    country: string;
  };
  measurementTime: string;
}
