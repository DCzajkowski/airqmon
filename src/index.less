@import "variables";
@import "common";
@import "tray-window/index";
@import "ErrorBoundary";

.header-arrow {
  position: absolute;
  top: 2px;
  left: 50%;
  margin-left: -5px;
  height: 10px;
  width: 10px;
  transform: rotate(45deg);
  background-color: #e8e6e8;
}

.window {
  position: absolute;
  top: 5px;

  .toolbar {
    box-shadow: none;
  }

  .toolbar-header {
    .title {
      margin-top: 2px;
    }
  }

  .window-content {
    background-color: @background-color;
  }
}

.window, .toolbar-header {
  border-top-left-radius: @window-border-radius;
  border-top-right-radius: @window-border-radius;
}
