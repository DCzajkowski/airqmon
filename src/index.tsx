import { injectGlobal } from './styled-components';
import * as React from 'react';
import { render } from 'react-dom';

import viewer from './analytics';
import { App } from './app';

const keys = require('../keys.json');

viewer.screenview('Tray window', 'Airqmon').send();

injectGlobal`
  @font-face {
    font-family: -apple-system, BlinkMacSystemFont, '.SFNSDisplay-Regular', 'Helvetica Neue', 'Lucida Grande', sans-serif;
    font-size: 1.1em;
    line-height: 1.5;
  }
`;

render(<App airlyToken={keys.airly} />, document.getElementById('app'));
