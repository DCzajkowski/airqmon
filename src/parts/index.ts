export { default as CenteredContent } from './CenteredContent';
export { default as ContentSpacing } from './ContentSpacing';
export { default as Link } from './Link';
export { Text, CenteredText } from './Text';
export { Icon, PhotonIcon } from './Icon';
export { default as Button } from './Button';
