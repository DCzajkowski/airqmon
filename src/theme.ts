export interface IThemeInterface {
  spacing: string;
  background: {
    primaryColor: string;
    darkColor: string;
  };
  updateAlert: {
    primaryBackgroundColor: string;
  };
  border: string;
  borderColor: string;
  borderRadius: string;
  text: {
    primarySize: string;
    secondarySize: string;
    smallSize: string;
    primaryColor: string;
    lightColor: string;
    darkColor: string;
    errorColor: string;
    airQualityIndexColor: {
      $1: string;
      $2: string;
      $3: string;
      $4: string;
      $5: string;
    };
  };
}

export const LightTheme: IThemeInterface = {
  spacing: '8px',
  background: {
    primaryColor: '#ffffff',
    darkColor: '#ececec',
  },
  updateAlert: {
    primaryBackgroundColor: '#dff0d8',
  },
  border: '1px solid #c2c0c2',
  borderColor: 'c2c0c2',
  borderRadius: '6px',
  text: {
    primarySize: '10pt',
    secondarySize: '8pt',
    smallSize: '7pt',
    primaryColor: '#333',
    lightColor: '#eaeaea',
    darkColor: '#666',
    errorColor: '#cc0000',
    airQualityIndexColor: {
      $1: '#79bc6a',
      $2: '#bbcf4c',
      $3: '#eec20b',
      $4: '#f29305',
      $5: '#e8416f',
    },
  },
};
