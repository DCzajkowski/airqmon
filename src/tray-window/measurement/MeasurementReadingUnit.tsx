import * as React from 'react';

export enum Unit {
  PM = 'PM',
  TEMP_C = 'TEMP_C',
  PRESSURE_PA = 'PRESSURE_PA',
  PERCENT = 'PERCENT',
}

export interface IMeasurementReadingUnitProps {
  unit?: Unit;
}

export const MeasurementReadingUnit = ({ unit = null }: IMeasurementReadingUnitProps) => {
  let content = {
    [Unit.PM]: (
      <>
        Î¼g/m<sup>3</sup>
      </>
    ),
    [Unit.TEMP_C]: (
      <>
        <sup>&deg;</sup>C
      </>
    ),
    [Unit.PRESSURE_PA]: 'hPA',
    [Unit.PERCENT]: '%',
  }[unit];

  if (content === undefined) {
    content = null;
  }

  return unit !== null ? <div className="measurement__unit"> {content}</div> : null;
};
