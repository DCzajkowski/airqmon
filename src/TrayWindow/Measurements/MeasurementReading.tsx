import * as React from 'react';

import { MeasurementReadingUnit, IMeasurementReadingUnitProps } from './MeasurementReadingUnit';

export interface IMeasurementReadingProps extends IMeasurementReadingUnitProps {
  reading: number|string;
  formatter?: (val: number|string) => string;
}

export const MeasurementReading = ({
  reading,
  formatter = null,
  unit = null,
}: IMeasurementReadingProps) => {
  const content = reading ? formatter ? formatter(reading) : reading : '-';

  return (
    <div className="reading">
      <span className="value">{content}</span>
      <MeasurementReadingUnit unit={unit} />
    </div>
  );
};
