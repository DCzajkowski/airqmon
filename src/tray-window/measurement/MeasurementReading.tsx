import * as React from 'react';

import { MeasurementReadingUnit, IMeasurementReadingUnitProps } from './MeasurementReadingUnit';

export interface IMeasurementReadingProps extends IMeasurementReadingUnitProps {
  reading?: number | string;
  formatter?: (val: number | string) => string;
}

export const MeasurementReading: React.SFC<IMeasurementReadingProps> = ({
  reading,
  formatter,
  unit,
  children,
}) => {
  const content = reading ? (formatter ? formatter(reading) : reading) : 'n/a';

  return (
    <>
      <div className="measurement__reading">
        {content}
        {reading ? <MeasurementReadingUnit unit={unit} /> : null}
        {children}
      </div>
    </>
  );
};
