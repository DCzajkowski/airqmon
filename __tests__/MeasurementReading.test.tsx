import { shallow } from 'enzyme';
import * as React from 'react';
import { LightTheme } from '../src/theme';
import MeasurementReading from '../src/tray-window/measurement/MeasurementReading';
import MeasurementReadingUnit, {
  Unit,
} from '../src/tray-window/measurement/MeasurementReadingUnit';
import { shallowWithTheme } from './test-helpers';
import { wrap } from 'module';
import { CenteredText } from '../src/parts';

function getWrapper({ props = {}, children = null } = {}) {
  return shallowWithTheme(
    <MeasurementReading {...props}>{children}</MeasurementReading>,
    LightTheme,
  ).dive();
}

describe('<MeasurementReading />', () => {
  const reading = 1.23;
  const formatter = jest.fn((val) => val.toFixed(0));

  afterEach(() => {
    formatter.mockClear();
  });

  it('renders correctly', () => {
    const wrapper = getWrapper({ props: { reading, formatter } });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(CenteredText)).toHaveLength(1);
  });

  it('calls provided formatter', () => {
    const wrapper = getWrapper({ props: { reading, formatter } });

    expect(formatter).toHaveBeenCalledTimes(1);
    expect(formatter).toHaveBeenCalledWith(reading);
  });

  it('renders correctly without formatter', () => {
    const wrapper = getWrapper({ props: { reading } });

    expect(wrapper).toMatchSnapshot();
  });

  it('renders passed children', () => {
    const wrapper = getWrapper({ props: { reading }, children: <div>A children node</div> });

    expect(wrapper).toMatchSnapshot();
  });

  it('Does not render <MeasurementReadingUnit /> if unit is not provided', () => {
    const wrapper = getWrapper({ props: { reading } });

    expect(wrapper.find(MeasurementReadingUnit)).toHaveLength(0);
  });

  it('Renders <MeasurementReadingUnit /> if unit is provided', () => {
    const wrapper = getWrapper({ props: { reading, unit: Unit.PRESSURE_PA } });

    expect(wrapper.find(MeasurementReadingUnit)).toHaveLength(1);
  });
});
