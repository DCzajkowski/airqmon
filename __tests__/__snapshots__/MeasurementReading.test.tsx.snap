// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeasurementReading /> renders correctly 1`] = `
.c0 > .sc-EHOje {
  font-size: calc(1em + 0.1em);
}

<div
  className="c0"
>
  <CenteredText>
    1
  </CenteredText>
</div>
`;

exports[`<MeasurementReading /> renders correctly without formatter 1`] = `
.c0 > .sc-EHOje {
  font-size: calc(1em + 0.1em);
}

<div
  className="c0"
>
  <CenteredText>
    1.23
  </CenteredText>
</div>
`;

exports[`<MeasurementReading /> renders passed children 1`] = `
.c0 > .sc-EHOje {
  font-size: calc(1em + 0.1em);
}

<div
  className="c0"
>
  <CenteredText>
    1.23
    <div>
      A children node
    </div>
  </CenteredText>
</div>
`;
